src/itemdef.h

```diff
@@ -76,8 +77,11 @@ struct ItemDefinition
 	u16 stack_max;
 	bool usable;
 	bool liquids_pointable;
-	// May be NULL. If non-NULL, deleted by destructor
+	
+	// They may be NULL. If non-NULL, deleted by destructor
+	PointingAbilities *pointabilities;
```

Comment:
I std::optional would be more appropriate here, since the main reason why this is a pointer is because it can be NULL.
However, carrying a const reference to an optional everywhere would a bit ugly, so making the optional property part of the PointingAbilities class would be good, also for the match functions.
But then every ItemDefinition has to store an unnecessary large PointingAbilities object, of course this can be solved by adding another abstraction layer.

Imo eventually the whole Minetest code should be upgraded to modern C++, but for now 
it can stay as a raw pointer, since ToolCapabilities also uses a raw pointer.